### PR TITLE
rename data pkg Stream to Item

### PR DESCRIPTION
**Pull request by ryanfkeepers**
A Stream is a continuous transmission of data.
An item is a single structure.  Crossing the two
definitions generates confusion.

Primarily code movement/renaming.  Though there
is also some reduction/replacement of structs
where we'd made a variety of testable Item implementations
instead of re-using the generic mock.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :broom: Tech Debt/Cleanup

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Review by ashmrtn — APPROVED** [review]
LGTM, can we wait to merge this until after the corso extension stuff goes in though? It's probably going to create merge conflicts so would be best to let the extensions stuff go in and resolve conflicts here. I think mostly it'll be #3987 that has them but I can't be certain

**Comment by pandeyabs** [comment]
Thanks @ashmrtn! @ryanfkeepers could you please hold off on this PR while the extension PRs go through?

**Comment by ryanfkeepers** [comment]
> LGTM, can we wait to merge this until after the corso extension stuff goes in though? It's probably going to create merge conflicts so would be best to let the extensions stuff go in and resolve conflicts here

Yep!  Not a problem.

File: src/internal/data/implementations.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+type FetchRestoreCollection struct {
+	Collection
+	FetchItemByNamer
+}
+
+// NoFetchRestoreCollection is a wrapper for a Collection that returns
+// ErrNotFound for all Fetch calls.
+type NoFetchRestoreCollection struct {
+	Collection
+	FetchItemByNamer
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/data/implementations.go:31]: I'll fix that in a follow-up PR.

File: src/internal/data/mock/collection.go [hunk 2]
```diff
@@ -14,46 +14,47 @@ import (
...
 
-type Stream struct {
-	ID           string
-	Reader       io.ReadCloser
-	ReadErr      error
-	ItemSize     int64
-	ModifiedTime time.Time
+type Item struct {
 	DeletedFlag  bool
+	Err          error
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/data/mock/collection.go:24]: Will do.

File: src/internal/data/implementations.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/data/implementations.go:31]: @ashmrtn please see: https://github.com/alcionai/corso/pull/4013

**Pull request merged**